### PR TITLE
Bump kotlin_labs definitions to v0.3

### PR DESCRIPTION
**Pull request by BoD**

### CODE REVIEW

File: docs/source/migration/4.0.mdx [hunk 1]
```diff
@@ -426,7 +426,7 @@ import data.builders.schema.__Schema
 
 ### KotlinLabs directives are now version 0.2
```
Comment (martinbonnin) [inline docs/source/migration/4.0.mdx:427]:
```suggestion
### KotlinLabs directives are now version 0.3
```

Comment (martinbonnin) [reply docs/source/migration/4.0.mdx:427]: Actually, this needs to be in the `next-docs` branch

File: libraries/apollo-ast/src/commonMain/kotlin/com/apollographql/apollo3/ast/gqldocument.kt [hunk 2]
```diff
@@ -80,14 +80,16 @@ fun builtinDefinitions() = definitionsFromString(builtinsDefinitionsStr)
...
+@ApolloInternal const val KOTLIN_LABS_VERSION = "v0.3"
 
 /**
  * Extra apollo Kotlin specific definitions from https://specs.apollo.dev/kotlin_labs/<[version]>
  */
 fun kotlinLabsDefinitions(version: String): List<GQLDefinition> {
   return definitionsFromString(when (version) {
-    KOTLIN_LABS_VERSION -> kotlinLabsDefinitions
+    // v0.3 has no behavior change over v0.2, so both versions are allowed as a convenience
+    KOTLIN_LABS_VERSION_0_2, KOTLIN_LABS_VERSION -> kotlinLabsDefinitions
```
Comment (martinbonnin) [inline libraries/apollo-ast/src/commonMain/kotlin/com/apollographql/apollo3/ast/gqldocument.kt:92]: 👍

File: docs/source/migration/4.0.mdx (hunk 1 above)
Comment (BoD) [reply docs/source/migration/4.0.mdx:427]: Woops good call! 👍

File: libraries/apollo-ast/src/commonMain/kotlin/com/apollographql/apollo3/ast/internal/definitions.kt [hunk 3]
```diff
@@ -15,59 +15,61 @@ internal val kotlinLabsDefinitions = """
...
+      ""${'"'}
+      embeddedFields: String! = "",
+      ""${'"'}
+      (experimental) a selection set containing fields that should be treated as [Relay Connection](https://relay.dev/graphql/connections.htm) fields.
+      Order is unimportant.
+      This works in conjunction with `ConnectionMetadataGenerator` and `ConnectionRecordMerger` which must be configured on the `ApolloStore`.
+      Since: 3.4.1
+      ""${'"'}
+      connectionFields: String! = ""
+  ) on OBJECT | INTERFACE
```
Comment (martinbonnin) [inline libraries/apollo-ast/src/commonMain/kotlin/com/apollographql/apollo3/ast/internal/definitions.kt:51]: Looks like a new IJ thing to use the continuation indent for parameter lists. Not a huge deal but i'm hoping we can stabilize this at some point.

**Review by martinbonnin — APPROVED** [review]

**Pull request merged**